python/paddle/fluid/layers/nn.py

```diff
@@ -4760,10 +4755,11 @@ def __check_input(x, y):
         if x_shape[-1] != y_shape[-2]:
             raise ValueError("Invalid inputs for matmul.")
 
-        if len(y_shape) > 2:
+        if len(y_shape) > 2 and len(x_shape) > 2:
```

Comment:
If the len(y_shape)  and `len(x_shape) > 2` are equal, the value of `y_shape[0:-2]` and `x_shape[0:-2]` should be equal.

Reply:
why